api/generated/common/types.go

```diff
@@ -339,6 +339,9 @@ type Block struct {
 	// \[txn\] TransactionsRoot authenticates the set of transactions appearing in the block. More specifically, it's the root of a merkle tree whose leaves are the block's Txids, in lexicographic order. For the empty block, it's 0. Note that the TxnRoot does not authenticate the signatures on the transactions, only the transactions themselves. Two blocks with the same transactions but in a different order and with different signatures will have the same TxnRoot.
 	TransactionsRoot []byte `json:"transactions-root"`
 
+	// \[txn256\] TransactionsRootSHA256 is an auxiliary TransactionRoot, built using a vector commitment instead of a merkle tree, and SHA256 hash function instead of the default SHA512_256. It is used mostly for StateProofs.
+	TransactionsRootSha256 []byte `json:"transactions-root-sha256"`
```

Comment:
should we give `TransactionsRootSha256` and `TransactionsRoot` the same names as in algod?

Reply:
I don't think algod defines the fields for a block--instead preferring to leave them as simply an opaque object which is returned. So in that sense these fields would not exist for algod to my knowledge.

Reply:
That field is not defined in `algod` as is, but it directly corresponds to the new fields in the `BlockHeader`.
I chose not to rename these fields (as I've renamed `TxnRoot -> TxnCommitments struct`) to stay backwards compatible to the previously defined type.

Reply:
We could rename them and keep the `json` keys as they are though;
```
TransactionsRoot -> TransactionCommitmentsNativeSha512_256
TransactionsRootSha256 -> TransactionCommitmentsSha256
```

Reply:
There is a field in the code generator which is supposed to allow us to diverge the json from the generated class name. However, it doesn't seem to be respected by the code generator when I change/run it, and also having the two diverge in that way seems confusing from a user perspective.

I think the path forward I'd like to take here is to just maintain the existing `transactions-root` and add in the `transactions-root-shaa256` as we are currently doing here.